devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java

```diff
@@ -241,6 +244,9 @@ public void execute() throws MojoFailureException, MojoExecutionException {
                     AppArtifact appArtifact = project.getAppArtifact();
                     MavenProject mavenProject = projectMap.get(String.format("%s:%s:%s",
                             appArtifact.getGroupId(), appArtifact.getArtifactId(), appArtifact.getVersion()));
+                    if (mavenProject == null) {
```

Comment:
This is bad. See https://github.com/quarkusio/quarkus/pull/2554

Reply:
If you take the sample app from #2498 and run mvn quarkus:dev from the 'runner' dir you will currently get a NPE (works fine when run from the root).